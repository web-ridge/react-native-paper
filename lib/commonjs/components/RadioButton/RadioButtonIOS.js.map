{"version":3,"sources":["RadioButtonIOS.tsx"],"names":["RadioButtonIOS","disabled","onPress","theme","status","value","testID","rest","checkedColor","colors","color","accent","rippleColor","text","alpha","rgb","string","fade","context","checked","contextValue","undefined","event","onValueChange","styles","container","opacity","displayName","StyleSheet","create","borderRadius","padding","React","memo","RadioButtonIOSWithTheme"],"mappings":";;;;;;;AAAA;;AACA;;AAMA;;AACA;;AACA;;AACA;;AACA;;AACA;;;;;;;;;;;;;;AAkCA;;;;;;;;;;;;;;;;AAgBA,MAAMA,cAAc,GAAG,UAQV;AAAA,MARW;AACtBC,IAAAA,QADsB;AAEtBC,IAAAA,OAFsB;AAGtBC,IAAAA,KAHsB;AAItBC,IAAAA,MAJsB;AAKtBC,IAAAA,KALsB;AAMtBC,IAAAA;AANsB,GAQX;AAAA,MADRC,IACQ;;AACX,QAAMC,YAAY,GAAGP,QAAQ,GACzBE,KAAK,CAACM,MAAN,CAAaR,QADY,GAEzBM,IAAI,CAACG,KAAL,IAAcP,KAAK,CAACM,MAAN,CAAaE,MAF/B;AAIA,MAAIC,WAAJ;;AAEA,MAAIX,QAAJ,EAAc;AACZW,IAAAA,WAAW,GAAG,oBAAMT,KAAK,CAACM,MAAN,CAAaI,IAAnB,EAAyBC,KAAzB,CAA+B,IAA/B,EAAqCC,GAArC,GAA2CC,MAA3C,EAAd;AACD,GAFD,MAEO;AACLJ,IAAAA,WAAW,GAAG,oBAAMJ,YAAN,EACXS,IADW,CACN,IADM,EAEXF,GAFW,GAGXC,MAHW,EAAd;AAID;;AAED,sBACE,oBAAC,oCAAD,CAAoB,QAApB,QACIE,OAAD,IAAsC;AACrC,UAAMC,OAAO,GACX,sBAAU;AACRC,MAAAA,YAAY,EAAEF,OAAF,aAAEA,OAAF,uBAAEA,OAAO,CAAEb,KADf;AAERD,MAAAA,MAFQ;AAGRC,MAAAA;AAHQ,KAAV,MAIO,SALT;AAOA,wBACE,oBAAC,wBAAD,eACME,IADN;AAEE,MAAA,UAAU,MAFZ;AAGE,MAAA,WAAW,EAAEK,WAHf;AAIE,MAAA,OAAO,EACLX,QAAQ,GACJoB,SADI,GAEHC,KAAD,IAAW;AACT,gCAAY;AACVpB,UAAAA,OADU;AAEVG,UAAAA,KAFU;AAGVkB,UAAAA,aAAa,EAAEL,OAAF,aAAEA,OAAF,uBAAEA,OAAO,CAAEK,aAHd;AAIVD,UAAAA;AAJU,SAAZ;AAMD,OAdT;AAgBE,MAAA,iBAAiB,EAAC,OAhBpB;AAiBE,MAAA,kBAAkB,EAAE;AAAErB,QAAAA,QAAF;AAAYkB,QAAAA;AAAZ,OAjBtB;AAkBE,MAAA,uBAAuB,EAAC,QAlB1B;AAmBE,MAAA,KAAK,EAAEK,MAAM,CAACC,SAnBhB;AAoBE,MAAA,MAAM,EAAEnB;AApBV,qBAsBE,oBAAC,iBAAD;AAAM,MAAA,KAAK,EAAE;AAAEoB,QAAAA,OAAO,EAAEP,OAAO,GAAG,CAAH,GAAO;AAAzB;AAAb,oBACE,oBAAC,8BAAD;AACE,MAAA,gBAAgB,EAAE,KADpB;AAEE,MAAA,IAAI,EAAC,OAFP;AAGE,MAAA,IAAI,EAAE,EAHR;AAIE,MAAA,KAAK,EAAEX,YAJT;AAKE,MAAA,SAAS,EAAC;AALZ,MADF,CAtBF,CADF;AAkCD,GA3CH,CADF;AA+CD,CAvED;;AAyEAR,cAAc,CAAC2B,WAAf,GAA6B,iBAA7B;;AAEA,MAAMH,MAAM,GAAGI,wBAAWC,MAAX,CAAkB;AAC/BJ,EAAAA,SAAS,EAAE;AACTK,IAAAA,YAAY,EAAE,EADL;AAETC,IAAAA,OAAO,EAAE;AAFA;AADoB,CAAlB,CAAf;;4BAOeC,KAAK,CAACC,IAAN,CAAW,wBAAUjC,cAAV,CAAX,C,EAEf;;;;AACA,MAAMkC,uBAAuB,GAAG,wBAAUlC,cAAV,CAAhC,C,CACA","sourcesContent":["import * as React from 'react';\nimport {\n  ColorValue,\n  GestureResponderEvent,\n  StyleSheet,\n  View,\n} from 'react-native';\nimport color from 'color';\nimport { RadioButtonContext, RadioButtonContextType } from './RadioButtonGroup';\nimport { handlePress, isChecked } from './utils';\nimport MaterialCommunityIcon from '../MaterialCommunityIcon';\nimport TouchableRipple from '../TouchableRipple/TouchableRipple';\nimport { withTheme } from '../../core/theming';\nimport type { $RemoveChildren } from '../../types';\n\ntype Props = $RemoveChildren<typeof TouchableRipple> & {\n  /**\n   * Value of the radio button\n   */\n  value: string;\n  /**\n   * Status of radio button.\n   */\n  status?: 'checked' | 'unchecked';\n  /**\n   * Whether radio is disabled.\n   */\n  disabled?: boolean;\n  /**\n   * Function to execute on press.\n   */\n  onPress?: (event: GestureResponderEvent) => void;\n  /**\n   * Custom color for radio.\n   */\n  color?: ColorValue;\n  /**\n   * @optional\n   */\n  theme: ReactNativePaper.Theme;\n  /**\n   * testID to be used on tests.\n   */\n  testID?: string;\n};\n\n/**\n * Radio buttons allow the selection a single option from a set.\n * This component follows platform guidelines for iOS, but can be used\n * on any platform.\n *\n * <div class=\"screenshots\">\n *   <figure>\n *     <img src=\"screenshots/radio-enabled.ios.png\" />\n *     <figcaption>Enabled</figcaption>\n *   </figure>\n *   <figure>\n *     <img src=\"screenshots/radio-disabled.ios.png\" />\n *     <figcaption>Disabled</figcaption>\n *   </figure>\n * </div>\n */\nconst RadioButtonIOS = ({\n  disabled,\n  onPress,\n  theme,\n  status,\n  value,\n  testID,\n  ...rest\n}: Props) => {\n  const checkedColor = disabled\n    ? theme.colors.disabled\n    : rest.color || theme.colors.accent;\n\n  let rippleColor: string;\n\n  if (disabled) {\n    rippleColor = color(theme.colors.text).alpha(0.16).rgb().string();\n  } else {\n    rippleColor = color(checkedColor as string)\n      .fade(0.32)\n      .rgb()\n      .string();\n  }\n\n  return (\n    <RadioButtonContext.Consumer>\n      {(context?: RadioButtonContextType) => {\n        const checked =\n          isChecked({\n            contextValue: context?.value,\n            status,\n            value,\n          }) === 'checked';\n\n        return (\n          <TouchableRipple\n            {...rest}\n            borderless\n            rippleColor={rippleColor}\n            onPress={\n              disabled\n                ? undefined\n                : (event) => {\n                    handlePress({\n                      onPress,\n                      value,\n                      onValueChange: context?.onValueChange,\n                      event,\n                    });\n                  }\n            }\n            accessibilityRole=\"radio\"\n            accessibilityState={{ disabled, checked }}\n            accessibilityLiveRegion=\"polite\"\n            style={styles.container}\n            testID={testID}\n          >\n            <View style={{ opacity: checked ? 1 : 0 }}>\n              <MaterialCommunityIcon\n                allowFontScaling={false}\n                name=\"check\"\n                size={24}\n                color={checkedColor}\n                direction=\"ltr\"\n              />\n            </View>\n          </TouchableRipple>\n        );\n      }}\n    </RadioButtonContext.Consumer>\n  );\n};\n\nRadioButtonIOS.displayName = 'RadioButton.IOS';\n\nconst styles = StyleSheet.create({\n  container: {\n    borderRadius: 18,\n    padding: 6,\n  },\n});\n\nexport default React.memo(withTheme(RadioButtonIOS));\n\n// @component-docs ignore-next-line\nconst RadioButtonIOSWithTheme = withTheme(RadioButtonIOS);\n// @component-docs ignore-next-line\nexport { RadioButtonIOSWithTheme as RadioButtonIOS };\n"]}