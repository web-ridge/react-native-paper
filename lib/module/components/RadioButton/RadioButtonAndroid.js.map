{"version":3,"sources":["RadioButtonAndroid.tsx"],"names":["React","Animated","View","StyleSheet","color","RadioButtonContext","handlePress","isChecked","TouchableRipple","withTheme","BORDER_WIDTH","RadioButtonAndroid","disabled","onPress","theme","value","status","testID","rest","current","borderAnim","useRef","Value","radioAnim","isFirstRendering","scale","animation","useEffect","setValue","timing","toValue","duration","useNativeDriver","start","checkedColor","colors","accent","uncheckedColor","text","alpha","dark","rgb","string","rippleColor","radioColor","context","checked","contextValue","fade","undefined","event","onValueChange","styles","container","radio","borderColor","borderWidth","absoluteFill","radioContainer","dot","backgroundColor","transform","displayName","create","borderRadius","alignItems","justifyContent","height","width","margin","RadioButtonAndroidWithTheme"],"mappings":";;;;;;AAAA,OAAO,KAAKA,KAAZ,MAAuB,OAAvB;AACA,SACEC,QADF,EAEEC,IAFF,EAGEC,UAHF,QAMO,cANP;AAOA,OAAOC,KAAP,MAAkB,OAAlB;AACA,SAASC,kBAAT,QAA2D,oBAA3D;AACA,SAASC,WAAT,EAAsBC,SAAtB,QAAuC,SAAvC;AACA,OAAOC,eAAP,MAA4B,oCAA5B;AACA,SAASC,SAAT,QAA0B,oBAA1B;AAsCA,MAAMC,YAAY,GAAG,CAArB;AAEA;;;;;;;;;;;;;;;;;AAgBA,MAAMC,kBAAkB,GAAG,UAQd;AAAA,MARe;AAC1BC,IAAAA,QAD0B;AAE1BC,IAAAA,OAF0B;AAG1BC,IAAAA,KAH0B;AAI1BC,IAAAA,KAJ0B;AAK1BC,IAAAA,MAL0B;AAM1BC,IAAAA;AAN0B,GAQf;AAAA,MADRC,IACQ;;AACX,QAAM;AAAEC,IAAAA,OAAO,EAAEC;AAAX,MAA0BpB,KAAK,CAACqB,MAAN,CAC9B,IAAIpB,QAAQ,CAACqB,KAAb,CAAmBZ,YAAnB,CAD8B,CAAhC;AAIA,QAAM;AAAES,IAAAA,OAAO,EAAEI;AAAX,MAAyBvB,KAAK,CAACqB,MAAN,CAC7B,IAAIpB,QAAQ,CAACqB,KAAb,CAAmB,CAAnB,CAD6B,CAA/B;AAIA,QAAME,gBAAgB,GAAGxB,KAAK,CAACqB,MAAN,CAAsB,IAAtB,CAAzB;AAEA,QAAM;AAAEI,IAAAA;AAAF,MAAYX,KAAK,CAACY,SAAxB;AAEA1B,EAAAA,KAAK,CAAC2B,SAAN,CAAgB,MAAM;AACpB;AACA,QAAIH,gBAAgB,CAACL,OAArB,EAA8B;AAC5BK,MAAAA,gBAAgB,CAACL,OAAjB,GAA2B,KAA3B;AACA;AACD;;AAED,QAAIH,MAAM,KAAK,SAAf,EAA0B;AACxBO,MAAAA,SAAS,CAACK,QAAV,CAAmB,GAAnB;AAEA3B,MAAAA,QAAQ,CAAC4B,MAAT,CAAgBN,SAAhB,EAA2B;AACzBO,QAAAA,OAAO,EAAE,CADgB;AAEzBC,QAAAA,QAAQ,EAAE,MAAMN,KAFS;AAGzBO,QAAAA,eAAe,EAAE;AAHQ,OAA3B,EAIGC,KAJH;AAKD,KARD,MAQO;AACLb,MAAAA,UAAU,CAACQ,QAAX,CAAoB,EAApB;AAEA3B,MAAAA,QAAQ,CAAC4B,MAAT,CAAgBT,UAAhB,EAA4B;AAC1BU,QAAAA,OAAO,EAAEpB,YADiB;AAE1BqB,QAAAA,QAAQ,EAAE,MAAMN,KAFU;AAG1BO,QAAAA,eAAe,EAAE;AAHS,OAA5B,EAIGC,KAJH;AAKD;AACF,GAxBD,EAwBG,CAACjB,MAAD,EAASI,UAAT,EAAqBG,SAArB,EAAgCE,KAAhC,CAxBH;AA0BA,QAAMS,YAAY,GAAGhB,IAAI,CAACd,KAAL,IAAcU,KAAK,CAACqB,MAAN,CAAaC,MAAhD;AACA,QAAMC,cAAc,GAClBnB,IAAI,CAACmB,cAAL,IACAjC,KAAK,CAACU,KAAK,CAACqB,MAAN,CAAaG,IAAd,CAAL,CACGC,KADH,CACSzB,KAAK,CAAC0B,IAAN,GAAa,GAAb,GAAmB,IAD5B,EAEGC,GAFH,GAGGC,MAHH,EAFF;AAOA,MAAIC,WAAJ,EAA6BC,UAA7B;AAEA,sBACE,oBAAC,kBAAD,CAAoB,QAApB,QACIC,OAAD,IAAsC;AACrC,UAAMC,OAAO,GACXvC,SAAS,CAAC;AACRwC,MAAAA,YAAY,EAAEF,OAAF,aAAEA,OAAF,uBAAEA,OAAO,CAAE9B,KADf;AAERC,MAAAA,MAFQ;AAGRD,MAAAA;AAHQ,KAAD,CAAT,KAIO,SALT;;AAOA,QAAIH,QAAJ,EAAc;AACZ+B,MAAAA,WAAW,GAAGvC,KAAK,CAACU,KAAK,CAACqB,MAAN,CAAaG,IAAd,CAAL,CAAyBC,KAAzB,CAA+B,IAA/B,EAAqCE,GAArC,GAA2CC,MAA3C,EAAd;AACAE,MAAAA,UAAU,GAAG9B,KAAK,CAACqB,MAAN,CAAavB,QAA1B;AACD,KAHD,MAGO;AACL+B,MAAAA,WAAW,GAAGvC,KAAK,CAAC8B,YAAD,CAAL,CACXc,IADW,CACN,IADM,EAEXP,GAFW,GAGXC,MAHW,EAAd;AAIAE,MAAAA,UAAU,GAAGE,OAAO,GAAGZ,YAAH,GAAkBG,cAAtC;AACD;;AAED,wBACE,oBAAC,eAAD,eACMnB,IADN;AAEE,MAAA,UAAU,MAFZ;AAGE,MAAA,WAAW,EAAEyB,WAHf;AAIE,MAAA,OAAO,EACL/B,QAAQ,GACJqC,SADI,GAEHC,KAAD,IAAW;AACT5C,QAAAA,WAAW,CAAC;AACVO,UAAAA,OADU;AAEVsC,UAAAA,aAAa,EAAEN,OAAF,aAAEA,OAAF,uBAAEA,OAAO,CAAEM,aAFd;AAGVpC,UAAAA,KAHU;AAIVmC,UAAAA;AAJU,SAAD,CAAX;AAMD,OAdT;AAgBE,MAAA,iBAAiB,EAAC,OAhBpB;AAiBE,MAAA,kBAAkB,EAAE;AAAEtC,QAAAA,QAAF;AAAYkC,QAAAA;AAAZ,OAjBtB;AAkBE,MAAA,uBAAuB,EAAC,QAlB1B;AAmBE,MAAA,KAAK,EAAEM,MAAM,CAACC,SAnBhB;AAoBE,MAAA,MAAM,EAAEpC;AApBV,qBAsBE,oBAAC,QAAD,CAAU,IAAV;AACE,MAAA,KAAK,EAAE,CACLmC,MAAM,CAACE,KADF,EAEL;AACEC,QAAAA,WAAW,EAAEX,UADf;AAEEY,QAAAA,WAAW,EAAEpC;AAFf,OAFK;AADT,OASG0B,OAAO,gBACN,oBAAC,IAAD;AAAM,MAAA,KAAK,EAAE,CAAC3C,UAAU,CAACsD,YAAZ,EAA0BL,MAAM,CAACM,cAAjC;AAAb,oBACE,oBAAC,QAAD,CAAU,IAAV;AACE,MAAA,KAAK,EAAE,CACLN,MAAM,CAACO,GADF,EAEL;AACEC,QAAAA,eAAe,EAAEhB,UADnB;AAEEiB,QAAAA,SAAS,EAAE,CAAC;AAAEpC,UAAAA,KAAK,EAAEF;AAAT,SAAD;AAFb,OAFK;AADT,MADF,CADM,GAYJ,IArBN,CAtBF,CADF;AAgDD,GApEH,CADF;AAwED,CAjID;;AAmIAZ,kBAAkB,CAACmD,WAAnB,GAAiC,qBAAjC;AAEA,MAAMV,MAAM,GAAGjD,UAAU,CAAC4D,MAAX,CAAkB;AAC/BV,EAAAA,SAAS,EAAE;AACTW,IAAAA,YAAY,EAAE;AADL,GADoB;AAI/BN,EAAAA,cAAc,EAAE;AACdO,IAAAA,UAAU,EAAE,QADE;AAEdC,IAAAA,cAAc,EAAE;AAFF,GAJe;AAQ/BZ,EAAAA,KAAK,EAAE;AACLa,IAAAA,MAAM,EAAE,EADH;AAELC,IAAAA,KAAK,EAAE,EAFF;AAGLJ,IAAAA,YAAY,EAAE,EAHT;AAILK,IAAAA,MAAM,EAAE;AAJH,GARwB;AAc/BV,EAAAA,GAAG,EAAE;AACHQ,IAAAA,MAAM,EAAE,EADL;AAEHC,IAAAA,KAAK,EAAE,EAFJ;AAGHJ,IAAAA,YAAY,EAAE;AAHX;AAd0B,CAAlB,CAAf;AAqBA,eAAevD,SAAS,CAACE,kBAAD,CAAxB,C,CAEA;;AACA,MAAM2D,2BAA2B,GAAG7D,SAAS,CAACE,kBAAD,CAA7C,C,CACA;;AACA,SAAS2D,2BAA2B,IAAI3D,kBAAxC","sourcesContent":["import * as React from 'react';\nimport {\n  Animated,\n  View,\n  StyleSheet,\n  GestureResponderEvent,\n  ColorValue,\n} from 'react-native';\nimport color from 'color';\nimport { RadioButtonContext, RadioButtonContextType } from './RadioButtonGroup';\nimport { handlePress, isChecked } from './utils';\nimport TouchableRipple from '../TouchableRipple/TouchableRipple';\nimport { withTheme } from '../../core/theming';\nimport type { $RemoveChildren } from '../../types';\n\ntype Props = $RemoveChildren<typeof TouchableRipple> & {\n  /**\n   * Value of the radio button\n   */\n  value: string;\n  /**\n   * Status of radio button.\n   */\n  status?: 'checked' | 'unchecked';\n  /**\n   * Whether radio is disabled.\n   */\n  disabled?: boolean;\n  /**\n   * Function to execute on press.\n   */\n  onPress?: (event: GestureResponderEvent) => void;\n  /**\n   * Custom color for unchecked radio.\n   */\n  uncheckedColor?: ColorValue;\n  /**\n   * Custom color for radio.\n   */\n  color?: ColorValue;\n  /**\n   * @optional\n   */\n  theme: ReactNativePaper.Theme;\n  /**\n   * testID to be used on tests.\n   */\n  testID?: string;\n};\n\nconst BORDER_WIDTH = 2;\n\n/**\n * Radio buttons allow the selection a single option from a set.\n * This component follows platform guidelines for Android, but can be used\n * on any platform.\n *\n * <div class=\"screenshots\">\n *   <figure>\n *     <img src=\"screenshots/radio-enabled.android.png\" />\n *     <figcaption>Enabled</figcaption>\n *   </figure>\n *   <figure>\n *     <img src=\"screenshots/radio-disabled.android.png\" />\n *     <figcaption>Disabled</figcaption>\n *   </figure>\n * </div>\n */\nconst RadioButtonAndroid = ({\n  disabled,\n  onPress,\n  theme,\n  value,\n  status,\n  testID,\n  ...rest\n}: Props) => {\n  const { current: borderAnim } = React.useRef<Animated.Value>(\n    new Animated.Value(BORDER_WIDTH)\n  );\n\n  const { current: radioAnim } = React.useRef<Animated.Value>(\n    new Animated.Value(1)\n  );\n\n  const isFirstRendering = React.useRef<boolean>(true);\n\n  const { scale } = theme.animation;\n\n  React.useEffect(() => {\n    // Do not run animation on very first rendering\n    if (isFirstRendering.current) {\n      isFirstRendering.current = false;\n      return;\n    }\n\n    if (status === 'checked') {\n      radioAnim.setValue(1.2);\n\n      Animated.timing(radioAnim, {\n        toValue: 1,\n        duration: 150 * scale,\n        useNativeDriver: true,\n      }).start();\n    } else {\n      borderAnim.setValue(10);\n\n      Animated.timing(borderAnim, {\n        toValue: BORDER_WIDTH,\n        duration: 150 * scale,\n        useNativeDriver: false,\n      }).start();\n    }\n  }, [status, borderAnim, radioAnim, scale]);\n\n  const checkedColor = rest.color || theme.colors.accent;\n  const uncheckedColor =\n    rest.uncheckedColor ||\n    color(theme.colors.text)\n      .alpha(theme.dark ? 0.7 : 0.54)\n      .rgb()\n      .string();\n\n  let rippleColor: ColorValue, radioColor: ColorValue;\n\n  return (\n    <RadioButtonContext.Consumer>\n      {(context?: RadioButtonContextType) => {\n        const checked =\n          isChecked({\n            contextValue: context?.value,\n            status,\n            value,\n          }) === 'checked';\n\n        if (disabled) {\n          rippleColor = color(theme.colors.text).alpha(0.16).rgb().string();\n          radioColor = theme.colors.disabled;\n        } else {\n          rippleColor = color(checkedColor as string)\n            .fade(0.32)\n            .rgb()\n            .string();\n          radioColor = checked ? checkedColor : uncheckedColor;\n        }\n\n        return (\n          <TouchableRipple\n            {...rest}\n            borderless\n            rippleColor={rippleColor}\n            onPress={\n              disabled\n                ? undefined\n                : (event) => {\n                    handlePress({\n                      onPress,\n                      onValueChange: context?.onValueChange,\n                      value,\n                      event,\n                    });\n                  }\n            }\n            accessibilityRole=\"radio\"\n            accessibilityState={{ disabled, checked }}\n            accessibilityLiveRegion=\"polite\"\n            style={styles.container}\n            testID={testID}\n          >\n            <Animated.View\n              style={[\n                styles.radio,\n                {\n                  borderColor: radioColor,\n                  borderWidth: borderAnim,\n                },\n              ]}\n            >\n              {checked ? (\n                <View style={[StyleSheet.absoluteFill, styles.radioContainer]}>\n                  <Animated.View\n                    style={[\n                      styles.dot,\n                      {\n                        backgroundColor: radioColor,\n                        transform: [{ scale: radioAnim }],\n                      },\n                    ]}\n                  />\n                </View>\n              ) : null}\n            </Animated.View>\n          </TouchableRipple>\n        );\n      }}\n    </RadioButtonContext.Consumer>\n  );\n};\n\nRadioButtonAndroid.displayName = 'RadioButton.Android';\n\nconst styles = StyleSheet.create({\n  container: {\n    borderRadius: 18,\n  },\n  radioContainer: {\n    alignItems: 'center',\n    justifyContent: 'center',\n  },\n  radio: {\n    height: 20,\n    width: 20,\n    borderRadius: 10,\n    margin: 8,\n  },\n  dot: {\n    height: 10,\n    width: 10,\n    borderRadius: 5,\n  },\n});\n\nexport default withTheme(RadioButtonAndroid);\n\n// @component-docs ignore-next-line\nconst RadioButtonAndroidWithTheme = withTheme(RadioButtonAndroid);\n// @component-docs ignore-next-line\nexport { RadioButtonAndroidWithTheme as RadioButtonAndroid };\n"]}