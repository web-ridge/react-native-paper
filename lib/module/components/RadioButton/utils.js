export const handlePress = ({
  onPress,
  value,
  onValueChange,
  event
}) => {
  onValueChange ? onValueChange(value) : onPress === null || onPress === void 0 ? void 0 : onPress(event);
};
export const isChecked = ({
  value,
  status,
  contextValue
}) => {
  if (contextValue !== undefined && contextValue !== null) {
    return contextValue === value ? 'checked' : 'unchecked';
  } else {
    return status;
  }
};
//# sourceMappingURL=utils.js.map