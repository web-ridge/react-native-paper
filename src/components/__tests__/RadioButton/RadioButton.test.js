import React from 'react';
import renderer from 'react-test-renderer';
import { RadioButtonContext } from '../../RadioButton/RadioButtonGroup';
import RadioButton from '../../RadioButton/RadioButton';

describe('RadioButton', () => {
  // afterEach(() => jest.resetModules());

  describe('on default platform', () => {
    beforeAll(() => {
      jest.mock('react-native', () => {
        const RN = jest.requireActual('react-native');

        RN.Platform = () => ({
          select: (objs) => objs.default,
        });

        return RN;
      });
    });

    it('renders properly', () => {
      const tree = renderer.create(<RadioButton value="first" />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('on ios platform', () => {
    beforeAll(() => {
      jest.mock('react-native', () => {
        const RN = jest.requireActual('react-native');

        RN.Platform = () => ({
          select: (objs) => objs.ios,
        });

        return RN;
      });
    });

    it('renders properly', () => {
      const tree = renderer.create(<RadioButton value="first" />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('when RadioButton is wrapped by RadioButtonContext.Provider', () => {
    it('renders properly', () => {
      const tree = renderer
        .create(
          <RadioButtonContext.Provider value="first" onValueChange={() => {}}>
            <RadioButton value="first" />
          </RadioButtonContext.Provider>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('RadioButton with custom testID', () => {
    it('renders properly', () => {
      const tree = renderer
        .create(<RadioButton value="first" testID={'custom:testID'} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
