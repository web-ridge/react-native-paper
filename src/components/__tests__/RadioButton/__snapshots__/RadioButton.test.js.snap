// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton RadioButton with custom testID renders properly 1`] = `
<View
  accessibilityComponentType="radiobutton_unchecked"
  accessibilityLiveRegion="polite"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessibilityTraits="button"
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
  testID="custom:testID"
>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#03dac4",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`RadioButton on default platform renders properly 1`] = `
<View
  accessibilityComponentType="radiobutton_unchecked"
  accessibilityLiveRegion="polite"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessibilityTraits="button"
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#03dac4",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`RadioButton on ios platform renders properly 1`] = `
<View
  accessibilityComponentType="radiobutton_unchecked"
  accessibilityLiveRegion="polite"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessibilityTraits="button"
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#03dac4",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`RadioButton when RadioButton is wrapped by RadioButtonContext.Provider renders properly 1`] = `
<View
  accessibilityComponentType="radiobutton_unchecked"
  accessibilityLiveRegion="polite"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessibilityTraits="button"
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#03dac4",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
